### PR TITLE
Extended store with paymentModule

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#6581

*Description of changes:*

* Extends store with existing payment module.
* Refactors references to payment module.
* Fixed the Setters for Applicant info.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/components/dialogs/cancel.vue [hunk 1]
```diff
@@ -64,11 +65,11 @@ export default class CancelDialog extends Mixins(NameRequestMixin, PaymentMixin)
   }
 
   private get isVisible (): boolean {
-    return PaymentModule[PaymentTypes.CANCEL_MODAL_IS_VISIBLE]
+    return this.$store.getters['cancelModalIsVisible']
```
Comment (severinbeauvais) [inline src/components/dialogs/cancel.vue:68]: So this is what you were talking about.

It feels too "low level" or detailed, but if it helps for now, ok.

Can you (re)use a type for the property name?

File: src/modules/payment/index.ts [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
-import { getVuexStore } from '@/store'
+import { Module, VuexModule } from 'vuex-module-decorators'
 import paymentStore from './store'
 
-const store = getVuexStore()
-
-@Module({ dynamic: true, namespaced: false, store, name: 'paymentModule' })
-export class PaymentModule extends VuexModule {
+@Module
+export default class PaymentModule extends VuexModule {
```
Comment (severinbeauvais) [inline src/modules/payment/index.ts:5]: I like it.

File: src/store/actions.ts [hunk 3]
```diff
@@ -1080,7 +1080,7 @@ export const submit: any = async ({ commit, getters }): Promise<any> => {
       }
       request = await putNameReservation(getters.getNrId)
     }
-    if (request) await paymentModule.togglePaymentModal(true)
+    if (request) await dispatch('togglePaymentModal', true)
```
Comment (severinbeauvais) [inline src/store/actions.ts:1083]: OK, tell me the different between dispatch and a commit or action?

File: src/store/index.ts [hunk 4]
```diff
@@ -22,6 +23,9 @@ export function getVuexStore () {
   Vue.use(Vuex)
 
   const store = new Vuex.Store<any>({
+    modules: {
+      payment: PaymentModule
```
Comment (severinbeauvais) [inline src/store/index.ts:27]: Oh, gotcha. Looks great!

**Review by severinbeauvais — APPROVED** [review]
Looks very nice so far. Way cleaner. Great job!

File: src/components/dialogs/cancel.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/dialogs/cancel.vue:68]: It does for sure. 

I bet we can, good idea!

File: src/store/actions.ts (hunk 3 above)
Comment (cameron-freshworks) [reply src/store/actions.ts:1083]: So, from what i've read..
-Dispatch: To call an action ( after the next tick ) - Asynchronous
- Commit: When you are ready to commit a value to store. - Synchronous 

That being said. We have these compound methods that are doing both... 
I'm not entirely sure they are 'ACTIONS' but it is probably partially true.

Comment (severinbeauvais) [reply src/store/actions.ts:1083]: Why did you choose dispatch here?

Comment (cameron-freshworks) [reply src/store/actions.ts:1082]: Because `togglePaymentModal` is an action within the payment module.

Comment (severinbeauvais) [reply src/store/actions.ts:1082]: Oh, so that's how you call an action in another module? Gotcha.

**Comment by severinbeauvais** [comment]
Let's merge this soon as it's getting difficult to see new changes vs previous changes.

**Pull request merged**